lib/metanorma/collection_construct_model.rb

```diff
@@ -0,0 +1,208 @@
+        return fileref unless @fileref_resolver
+
+        @fileref_resolver.call(ref_folder, fileref)
+      end
+
+      # @param collection_model [Hash{String=>String}]
+      def compile_adoc_documents(collection_model)
+        documents = select_documents(collection_model)
+        return unless documents
+
+        documents["docref"]
+          .select { |k, _v| File.extname(k["fileref"]) == ".adoc" }
+          .each do |dr|
+          compile_adoc_file(dr["fileref"])
+          dr["fileref"] = set_adoc2xml(dr["fileref"])
+        end
+      end
+
+      # @param collection_model [Hash{String=>String}]
+      def select_documents(collection_model)
+        collection_model["manifest"]["manifest"]
+          .select { |k, _v| k["level"] == "document" }.first
+      end
+
+      # @param fileref [String]
+      def set_adoc2xml(fileref)
+        File.join(
+          File.dirname(fileref),
+          File.basename(fileref).gsub(/.adoc$/, ".xml"),
+        )
+      end
+
+      # param filepath [String]
+      # @raise [AdocFileNotFoundException]
+      def compile_adoc_file(filepath)
+        unless File.exist? filepath
+          raise AdocFileNotFoundException.new "#{filepath} not found!"
+        end
+
+        Util.log("[metanorma] Info: Compiling #{filepath}...", :info)
+
+        Metanorma::Compile.new
+          .compile(filepath, agree_to_terms: true, no_install_fonts: true)
+
+        Util.log("[metanorma] Info: Compiling #{filepath}...done!", :info)
+      end
+
+      # @param collection_model [Hash{String=>String}]
+      # @return [Hash{String=>String}]
+      def construct_collection_manifest(collection_model)
+        mnf = collection_model["manifest"]
+
+        mnf["docref"].each do |dr|
+          check_file_existance(dr["file"])
+          set_default_manifest(mnf)
+          construct_docref(mnf, dr)
+        end
+
+        # remove keys in upper level
+        mnf.delete("docref")
+        mnf.delete("sectionsplit")
+
+        collection_model
+      end
+
+      # @param filepath
+      # @raise [FileNotFoundException]
+      def check_file_existance(filepath)
+        unless File.exist?(filepath)
+          error_message = "#{filepath} not found!"
+          Util.log("[metanorma] Error: #{error_message}", :error)
+          raise FileNotFoundException.new error_message.to_s
+        end
+      end
+
+      # @param manifest [Hash{String=>String}]
+      def set_default_manifest(manifest)
+        manifest["manifest"] ||= [{ "level" => "document",
+                                    "title" => "Document",
+                                    "docref" => [] },
+                                  { "level" => "attachments",
+                                    "title" => "Attachments",
+                                    "docref" => [] }]
+      end
+
+      # @param collection_model [Hash{String=>String}]
+      # @return [Bool]
+      def new_yaml_format?(collection_model)
```

Comment:
Fixed at f75807acde632f6a5d2994861b1e1aa1df7fc348

Reply:
In case @ronaldtse protests the change to fileref, what I'm suggesting here is that (a) you have some other criterion for differentiating the old from new manifest, but more importantly (b) you represent the information conveyed with fileref in the old manifest, and file in the new manifest, in the same way—i.e. by changing the key on ingest, either so that it is only fileref, or so that it is only file. 

There is no reason not to preserve backward compatibility, if it is simply a matter of mapping the key on ingest (e.g. `manifest.transform_keys { |k| k == "fileref" and k = "file" }`, done recursively). Ronald may not care about breaking backward compatibility, but this is not the instance where it needs to be broken. And if you don't do this transform, then code that works with the new format no longer works with the old format, or vice versa.